src/components/Swap/SwapButton/index.tsx

```diff
@@ -118,11 +123,19 @@ export default memo(function SwapButton({ disabled }: SwapButtonProps) {
       })
       setDisplayTxHash(transaction.hash)
 
-      // Set the block containing the response to the oldest valid block to ensure that the
-      // completed trade's impact is reflected in future fetched trades.
-      transaction.wait(1).then((receipt) => {
-        setOldestValidBlock(receipt.blockNumber)
-      })
+      transaction
+        .wait(1)
+        .then((receipt) => {
+          if (receipt?.status === 0) {
+            onTxFail?.(new Error('Transaction failed'), receipt)
+          } else {
+            onTxSuccess?.(receipt.transactionHash, receipt)
+          }
+          // Set the block containing the response to the oldest valid block to ensure that the
+          // completed trade's impact is reflected in future fetched trades.
+          setOldestValidBlock(receipt.blockNumber)
+        })
+        .catch((e) => onTxFail?.(e, transaction.hash))
```

Comment:
If a transaction is cancelled by the user, it gets caught here. Does that count as a tx failing?